### PR TITLE
webgui 500 - AttributeError: 'AnonymousUserMixin' object has no attri…

### PR DESCRIPTION
**Pull request by doomedraven**
…bute 'has_role'

fixing this it will redirect to login page instead of give error 500

```
 File "yeti/core/auth/local/views.py", -> new_user
 if current_user.has_role('admin') and current_user.is_active:
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/local.py", line 347, in __getattr__
    return getattr(self._get_current_object(), name)
AttributeError: 'AnonymousUserMixin' object has no attribute 'has_role'
```

### CODE REVIEW

**Comment by doomedraven** [comment]
like this? `if type(current_user) != AnonymousUserMixin`

File: core/auth/local/views.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def logout():
 def new_user():
     username = request.form.get("username")
     password = request.form.get("password")
-    if current_user.has_role('admin') and current_user.is_active:
+    if type(current_user) != AnonymousUserMixin and current_user.has_role('admin') and current_user.is_active:
```
Comment (p-l-) [inline core/auth/local/views.py]: Type checks should use `isinstance()`, or `is` / `is not` operators (for strict type checks). In this case, I would suggest:
```suggestion
    if not isinstance(current_user, AnonymousUserMixin) and current_user.has_role('admin') and current_user.is_active:
```
See [this SO answer](https://stackoverflow.com/a/152596/3223422) for more.

**Comment by doomedraven** [comment]
thanks @p-l-

**Comment by gaelmuller** [comment]
I will add my own suggestion. I think that using `@login_required` would be much better.

**Pull request merged**